packages/cart/src/services/cart-module.ts

```diff
@@ -175,4 +175,105 @@ export default class CartModuleService implements ICartModuleService {
     const cartIds = Array.isArray(ids) ? ids : [ids]
     await this.cartService_.delete(cartIds, sharedContext)
   }
+
+  @InjectManager("baseRepository_")
+  async listAddresses(
+    filters: FilterableAddressProps = {},
+    config: FindConfig<AddressDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const addresses = await this.addressService_.list(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<CartAddressDTO[]>(addresses, {
+      populate: true,
+    })
+  }
+
+  async createAddresses(data: CreateAddressDTO, sharedContext?: Context)
+  async createAddresses(data: CreateAddressDTO[], sharedContext?: Context)
+
+  @InjectManager("baseRepository_")
+  async createAddresses(
+    data: CreateAddressDTO[] | CreateAddressDTO,
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const input = Array.isArray(data) ? data : [data]
+    const addresses = await this.createAddresses_(input, sharedContext)
+
+    const result = await this.listAddresses(
+      { id: addresses.map((p) => p.id) },
+      {},
+      sharedContext
+    )
+
+    return (Array.isArray(data) ? result : result[0]) as
+      | AddressDTO
+      | AddressDTO[]
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  protected async createAddresses_(
+    data: CreateAddressDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const addresses = await this.addressService_.create(data, sharedContext)
+
+    return await this.listAddresses(
+      { id: addresses.map((p) => p!.id) },
+      {},
+      sharedContext
+    )
```

Comment:
question: shouldn't this method just return the created address without calling list?

Reply:
I would say the return type of methods in the module service should always be the DTOs made specifically for the purpose. Otherwise, the return type would be the entity, unless I cast it.

What do you think?

Reply:
Oh wiat sorry, now I see what you mean. We should only list the addresses in the public method. That's an oversight from my side.

Reply:
yes thats what I was thinking but just wanted to be sure to bot miss anything